From d0e4b33b4a8d30c26c9740046a012295c1270ea9 Mon Sep 17 00:00:00 2001
From: Li Jun <jun.li@nxp.com>
Date: Fri, 17 Aug 2018 10:51:38 +0800
Subject: [PATCH 4418/5242] MLK-19254 usb: cdns3: gadget: set gadget state to
 be USB_STATE_NOTATTACHED after stop

commit  27a64bb9ca48a9422c1ed6af91201c96107726aa from
https://source.codeaurora.org/external/imx/linux-imx.git

Update gadget state to be USB_STATE_NOTATTACHED after stop gaget.

Signed-off-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/cdns3/gadget.c |    1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/cdns3/gadget.c b/drivers/usb/cdns3/gadget.c
index b3d81aa..c01cbf5 100644
--- a/drivers/usb/cdns3/gadget.c
+++ b/drivers/usb/cdns3/gadget.c
@@ -2460,6 +2460,7 @@ static void __cdns3_gadget_stop(struct cdns3 *cdns)
 	usb_gadget_disconnect(&usb_ss->gadget);
 	spin_lock_irqsave(&usb_ss->lock, flags);
 	usb_ss->gadget.speed = USB_SPEED_UNKNOWN;
+	usb_gadget_set_state(&usb_ss->gadget, USB_STATE_NOTATTACHED);
 	/* disable interrupt for device */
 	gadget_writel(usb_ss, &usb_ss->regs->usb_ien, 0);
 	gadget_writel(usb_ss, &usb_ss->regs->usb_conf, USB_CONF__DEVDS__MASK);
-- 
1.7.9.5

